### ISSUE TITLE
NXLoader SUMBITURB failed :(

### ISSUE THREAD

**Issue by theoouy** [DavidBuchanan314/NXLoader#11]
Like the title says, I can't seem to complete the run because of this weird error, and it's driving me nuts.  I even used different cords, and 2 different phones.  Does anyone have any idea what this error might consist of?

**Comment by a7mag3ddon** [comment]
using a usb -c on the phone? i read it needs usb 3.0 to work.

**Comment by DavidBuchanan314** [comment]
@a7mag3ddon not true. Where did you read that?

SUBMITURB failed most likely means your device has an EHCI controller, not XHCI. You'd need to patch your kernel's drivers if you want this to work.

**Comment by a7mag3ddon** [comment]
I thought I had read that it uses xhci chipset aka usb 3.0.

Will check my history and report back.

**Comment by a7mag3ddon** [comment]
Hang on, just saw your comment above, saying the same thing lol

**Comment by a7mag3ddon** [comment]
Just googled switch hack xhci there are loads of sites saying it, that's where I would have got it. Probably causing his problem as you've said above.

**Comment by DavidBuchanan314** [comment]
@a7mag3ddon XHCI != USB 3.0. 

My nexus 5 is USB 2.0 only, yet it has an XHCI controller.

**Comment by theoouy** [comment]
And you'd think that an LG Stylo 3 Plus has one.

**Comment by a7mag3ddon** [comment]
according to the specs LG Stylo 3 Plus is micro usb 2.0 i dont know what controller it has in it doesnt say. 

Fail0ver said that usb 2.0 driver on xhci will need a kernal patch which is available from their page, as my PC and phone are usb 3.0 seems to work find for me :-/

**Comment by DavidBuchanan314** [comment]
@a7mag3ddon xhci does not need patching under any circumstances, ehci does however.

**Comment by theoouy** [comment]
Yeah, I just did it with my friends phone which is a Samsung Galaxy 7 and it worked, so I'm assuming that if your phone can use a usb driver but can't run it, you need to kernel patch it, otherwise it's dandy for other phones.

**Comment by aKMTR** [comment]
But how do i patch my kernel to xhci? I been looking all over the internet and i could barely find info on how to patch the kernel to xhci

**Comment by aKMTR** [comment]
@DavidBuchanan314

**Comment by DavidBuchanan314** [comment]
@aKMTR not patch *to* xhci, simply patch *ehci* to behave the way we want it to.

https://gist.github.com/DavidBuchanan314/41b12362cc4d8c539dc441d75155f2e9

**Comment by gotwig** [comment]
GALAXY S6 working!!! use the usb hub with the name MICRO USB HUB on it that looks exactly like this https://rover.ebay.com/rover/0/0/0?mpre=https%3A%2F%2Fwww.ebay.de%2Fulk%2Fitm%2F131526067980

in different countries its everywhere selling, look for the name on the hub and search for the same picture!! good luck and thank you discord!

**Comment by mariogamer2** [comment]
Do you need to rebuild the whole kernel or only the driver?

**Comment by DavidBuchanan314** [comment]
Depends if it's built as a module or not...

**Comment by Zacki777** [comment]
how can i patch kernel? i  have redmi note 3 pro with nitrogen os anfroid 7.1.2? same problem for me(

**Comment by symis** [comment]
DavidBuchanan314 please help us on patching the kernel and solve this problem. A small guide would be really appreciated. Thanks

**Comment by walkthetalk** [comment]
Can we resolve it though patching nxloader? generally speaking, we can't patch the kernel for android if not rooted.
